File: dotnet/GitHub/Repos/Item/Item/Commits/Item/WithCommit_shaItemRequestBuilder.cs
using GitHubTodoDemo.GitHub.Repos.Item.Item.Commits.Item.Pulls;
using Microsoft.Kiota.Abstractions;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using System;
namespace GitHubTodoDemo.GitHub.Repos.Item.Item.Commits.Item {
    /// <summary>
    /// Builds and executes requests for operations under \repos\{owner}\{repo}\commits\{commit_sha}
    /// </summary>
    public class WithCommit_shaItemRequestBuilder : BaseRequestBuilder {
        /// <summary>The pulls property</summary>
        public PullsRequestBuilder Pulls { get =>
            new PullsRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>
        /// Instantiates a new WithCommit_shaItemRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="pathParameters">Path parameters for the request</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public WithCommit_shaItemRequestBuilder(Dictionary<string, object> pathParameters, IRequestAdapter requestAdapter) : base(requestAdapter, "{+baseurl}/repos/{owner}/{repo}/commits/{commit_sha}", pathParameters) {
        }
        /// <summary>
        /// Instantiates a new WithCommit_shaItemRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="rawUrl">The raw URL to use for the request builder.</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public WithCommit_shaItemRequestBuilder(string rawUrl, IRequestAdapter requestAdapter) : base(requestAdapter, "{+baseurl}/repos/{owner}/{repo}/commits/{commit_sha}", rawUrl) {
        }
    }
}

File: php/src/GitHub/Repos/Item/Item/Commits/Item/WithCommit_shaItemRequestBuilder.php
<?php

namespace Baywet\Githubtododemo\Github\Repos\Item\Item\Commits\Item;

use Baywet\Githubtododemo\Github\Repos\Item\Item\Commits\Item\Pulls\PullsRequestBuilder;
use Microsoft\Kiota\Abstractions\BaseRequestBuilder;
use Microsoft\Kiota\Abstractions\RequestAdapter;

/**
 * Builds and executes requests for operations under /repos/{owner}/{repo}/commits/{commit_sha}
*/
class WithCommit_shaItemRequestBuilder extends BaseRequestBuilder 
{
    /**
     * The pulls property
    */
    public function pulls(): PullsRequestBuilder {
        return new PullsRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * Instantiates a new WithCommit_shaItemRequestBuilder and sets the default values.
     * @param array<string, mixed>|string $pathParametersOrRawUrl Path parameters for the request or a String representing the raw URL.
     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
    */
    public function __construct($pathParametersOrRawUrl, RequestAdapter $requestAdapter) {
        parent::__construct($requestAdapter, [], '{+baseurl}/repos/{owner}/{repo}/commits/{commit_sha}');
        if (is_array($pathParametersOrRawUrl)) {
            $this->pathParameters = $pathParametersOrRawUrl;
        } else {
            $this->pathParameters = ['request-raw-url' => $pathParametersOrRawUrl];
        }
    }

}
